eng/common/scripts/Package-Properties.ps1

```diff
@@ -77,16 +77,43 @@ function Get-PkgProperties
     (
         [Parameter(Mandatory = $true)]
         [string]$PackageName,
-        [Parameter(Mandatory = $true)]
         [string]$ServiceDirectory
     )
 
-    $pkgDirectoryPath = $null
-    $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $ServiceDirectory
-    if (!(Test-Path $serviceDirectoryPath))
+    if ($ServiceDirectory)
+    {
+        $serviceDirectoryPath = Join-Path $RepoRoot "sdk" $ServiceDirectory
+        if (!(Test-Path $serviceDirectoryPath))
+        {
+            LogError "Service Directory [ $ServiceDirectory ] does not exist"
+            return $null
+        }
+    }
+    else
     {
-        LogError "Service Directory $ServiceDirectory does not exist"
-        return $null
+        $searchDir = Join-Path $RepoRoot "sdk"
+        foreach ($dir in (Get-ChildItem $searchDir -Directory))
+        {
+            $serviceDirectoryPath = Join-Path $searchDir $dir.Name
+            $ciYmlPath = Join-Path $serviceDirectoryPath "ci.yml"
+
+            if (Test-Path $ciYmlPath)
+            {
+                $activePkgList = Get-PkgListFromYml -ciYmlPath $ciYmlPath
+                $pkgInfoFromYml = $activePkgList | Where-Object { $_["name"] -eq $PackageName }
```

Comment:
I don't think this is going to work in languages like JS and Python. As what we have here isn't actually the package name but the artifact name. 

For example JS: https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/storage/ci.yml#L29 the package name is `@azure\storage-blob". We hacked around this in the package info function at https://github.com/Azure/azure-sdk-for-js/blob/master/eng/scripts/Language-Settings.ps1#L16 but I don't think that will work in our case. 

@praveenkuttappan and I were working through this issue with other work he was working on. See https://github.com/Azure/azure-sdk-for-js/pull/13542/files for some more details. 

Instead I think we probably should start moving these package property functions such that we can simply read all projects for a given service directory or the root of the sdk folder. Then we can actually use the properties return to compare the real properties instead of having to hack around.

Reply:
I did the filtering directly in the `Get-AllPkgProperties`